packages/rum-core/src/domain/assembly.ts

```diff
@@ -81,7 +81,7 @@ export function startRumAssembly(
     ({ startTime, rawRumEvent, domainContext, savedCommonContext, customerContext }) => {
       const viewContext = parentContexts.findView(startTime)
       const urlContext = urlContexts.findUrl(startTime)
-      const session = sessionManager.findTrackedSession(startTime)
+      const session = sessionManager.findTrackedSession(rawRumEvent.type !== RumEventType.VIEW ? startTime : undefined)
```

Comment:
When looking at the `assembly` code alone, this seems risky. It indicates that a View could potentially be associated to multiple sessions when updates are sent before and after a session renew. What about adding a comment to explain:

* why are we using the current session for view events (why are we making this exception)
* why this is fine and view events are only generated for the current session (contrary to other events)

Reply:
added some notes + creating a task for the todo